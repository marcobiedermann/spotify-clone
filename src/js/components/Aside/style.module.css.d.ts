declare const styles: {
  readonly aside: string;
  readonly 'aside--left': string;
};
export = styles;
