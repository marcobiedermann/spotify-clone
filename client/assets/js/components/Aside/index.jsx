import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.css';

class Aside extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const { children } = this.props;

    return (
      <aside className={classNames(styles.aside, styles['aside--left'])}>
        {children}
      </aside>
    );
  }
}

export default Aside;
