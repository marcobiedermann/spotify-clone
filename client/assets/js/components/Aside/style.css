.aside {
  flex-shrink: 0;
  min-width: 14em;
}

.aside--left {
  order: -1;
}
