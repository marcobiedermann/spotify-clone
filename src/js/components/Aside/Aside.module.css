.aside {
  background-color: #000;
  flex-shrink: 0;
  width: 14em;
}

.aside--left {
  order: -1;
}
