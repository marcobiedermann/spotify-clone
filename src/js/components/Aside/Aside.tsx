import clsx from 'clsx';
import React, { ReactNode } from 'react';
import styles from './Aside.module.css';

export interface AsideProps {
  children: ReactNode;
  className?: string;
}

function Aside(props: AsideProps): JSX.Element {
  const { className, ...otherProps } = props;

  return <aside className={clsx(className, styles.aside, styles['aside--left'])} {...otherProps} />;
}

export default Aside;
