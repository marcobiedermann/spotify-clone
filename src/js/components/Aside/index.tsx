import classNames from 'classnames';
import React, { FC } from 'react';
import styles from './style.module.css';

export interface AsideProps {
  className?: string;
}

const Aside: FC<AsideProps> = (props) => {
  const { className, ...otherProps } = props;

  return (
    <aside className={classNames(className, styles.aside, styles['aside--left'])} {...otherProps} />
  );
};

export default Aside;
