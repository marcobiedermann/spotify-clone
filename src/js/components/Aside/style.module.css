.aside {
  background-color: #121212;
  flex-shrink: 0;
  padding-left: 1.5em;
  padding-right: 1.5em;
  width: 14em;
}

.aside--left {
  order: -1;
}
