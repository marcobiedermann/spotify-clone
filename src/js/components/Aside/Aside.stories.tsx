import type { Meta, StoryObj } from '@storybook/react';
import Aside from '.';

const meta: Meta<typeof Aside> = {
  component: Aside,
  args: {
    children: 'Aside',
  },
};

type Story = StoryObj<typeof Aside>;

const Default: Story = {};

export { Default };
export default meta;
