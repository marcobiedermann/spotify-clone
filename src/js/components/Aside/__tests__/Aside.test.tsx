import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';
import Aside from '..';

it('renders correctly', () => {
  const { container } = render(<Aside>Aside</Aside>);

  expect(container.firstChild).toMatchSnapshot();
});
