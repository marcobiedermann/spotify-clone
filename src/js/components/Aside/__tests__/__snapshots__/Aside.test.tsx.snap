// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<aside
  class="aside aside--left"
>
  Aside
</aside>
`;
